src/librustc_mir/build/matches/test.rs

```diff
@@ -344,7 +344,8 @@ impl<'a, 'gcx, 'tcx> Builder<'a, 'gcx, 'tcx> {
                         func: Operand::Constant(box Constant {
                             span: test.span,
                             ty: mty,
-                            literal: method
+                            user_ty: None, // FIXME
```

Comment:
can you elaborate (perhaps in this FIXME comment, or perhaps just in a response to me here) on which kind of input expressions you expect need to reflect a user-type annotation? If I understand this part of the source correctly, this is where we are generating code to compare a pattern literal to the correspond part of the `match` input; does Rust currently have a way to express a user-written type annotation on such a pattern literal? Or are you just anticipating https://github.com/rust-lang/rfcs/issues/354 or https://github.com/rust-lang/rfcs/pull/2522 ?

Reply:
TBH I don't know, but it was a constant that derived from user input. Can we use e.g. associated constants in match? If so, then this might apply, since I guess you could do something like `<T as Trait<'a>>::BAR` (also, I'm not entirely how we should handle that, either. But that's for the next PR. =)